src/pymor/vectorarrays/list.py

```diff
@@ -342,211 +341,186 @@ def conj(self):
             R = self.space.zero_vector()
-            for v, c in zip(self._list, coeffs):
+            for v, c in zip(self._indexed(ind), coeffs):
                 R.axpy(c, v)
             RL.append(R)
 
-        return ListVectorArray(RL, self.space)
+        return ListVectorArrayImpl(RL, self.space)
 
-    def _norm(self):
-        return np.array([v.norm() for v in self._list])
+    def norm(self, ind):
+        return np.array([v.norm() for v in self._indexed(ind)])
 
-    def _norm2(self):
-        return np.array([v.norm2() for v in self._list])
+    def norm2(self, ind):
+        return np.array([v.norm2() for v in self._indexed(ind)])
 
-    def sup_norm(self):
-        if self.dim == 0:
-            return np.zeros(len(self))
-        else:
-            return np.array([v.sup_norm() for v in self._list])
-
-    def dofs(self, dof_indices):
-        assert isinstance(dof_indices, list) and (len(dof_indices) == 0 or min(dof_indices) >= 0) \
-            or (isinstance(dof_indices, np.ndarray) and dof_indices.ndim == 1
-                and (len(dof_indices) == 0 or np.min(dof_indices) >= 0))
-        assert len(self) > 0 or len(dof_indices) == 0 or max(dof_indices) < self.dim
-        return np.array([v.dofs(dof_indices) for v in self._list]).reshape((len(self), len(dof_indices)))
-
-    def amax(self):
-        assert self.dim > 0
+    def dofs(self, dof_indices, ind):
+        return (np.array([v.dofs(dof_indices) for v in self._indexed(ind)])
+                  .reshape((self.len_ind(ind), len(dof_indices))))
 
-        MI = np.empty(len(self._list), dtype=int)
-        MV = np.empty(len(self._list))
+    def amax(self, ind):
+        l = self.len_ind(ind)
+        MI = np.empty(l, dtype=int)
+        MV = np.empty(l)
 
-        for k, v in enumerate(self._list):
+        for k, v in enumerate(self._indexed(ind)):
             MI[k], MV[k] = v.amax()
 
         return MI, MV
 
-    @property
-    def real(self):
-        return ListVectorArray([v.real for v in self._list], self.space)
+    def real(self, ind):
+        return ListVectorArrayImpl([v.real for v in self._indexed(ind)], self.space)
 
-    @property
-    def imag(self):
+    def imag(self, ind):
         # note that Vector.imag is allowed to return None in case
         # of a real vector, so we have to check for that.
         # returning None is allowed as ComplexifiedVector does not know
         # how to create a new zero vector.
-        return ListVectorArray([v.imag or self.space.zero_vector() for v in self._list], self.space)
+        return ListVectorArrayImpl([v.imag or self.space.zero_vector() for v in self._indexed(ind)], self.space)
+
+    def conj(self, ind):
+        return ListVectorArrayImpl([v.conj() for v in self._indexed(ind)], self.space)
 
-    def conj(self):
-        return self.__class__([v.conj() for v in self._list], self.space)
+
+class ListVectorArray(VectorArray):
+    """|VectorArray| implemented as a Python list of vectors.
```

Comment:
Refer to the `Impl` here too?

Reply:
For similar reasons as above, I'd prefer to not do it.